### PR TITLE
New Web Background Sync documentation

### LINKED ISSUE #8904: Content suggestion: Web Background Sync
## What is the new suggestion?
While MDN has a page for the `Web Periodic Background Sync` API, it doesn't for the (older) `Web Background Sync` API.
I'm proposing to add a new page for the latter, similar to the former.

## Why is it important or useful?
Those APIs are named similarly, but are still different and serve different use cases.
They are both implemented in (at least) chromium.

People building PWAs most likely will be interested in the Background Sync API as it allows apps to sync short bits of data with the server in the background (i.e. without the user using the app) when a connection is established. This is useful for apps to provide a robust offline experience.

I think this should be mostly 1 page, based on the existing Periodic Background Sync page (very similar in fact), with a few links to be fixed here and there I guess.

### PR DESCRIPTION
**Pull request by captainbrosset**
This fixes #8904. A page about the `Web Background Sync` API was missing. This PR is adding it.

### CODE REVIEW

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: Web Background Synchronization API
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:2]:
```suggestion
title: Background Synchronization API
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+slug: Web/API/Web_Background_Synchronization_API
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:3]:
```suggestion
slug: Web/API/Background_Synchronization_API
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  - PWA
+  - Service Workers
+  - Sync
+  - Web Background Synchronization API
+---
+{{securecontext_header}}
+
+{{DefaultAPISidebar("Background Sync")}}
+
+The Web Background Synchronization API provides a way to defer tasks to be run in a {{domxref('Service Worker API','service worker')}} until the user has a stable network connection.
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:19]:
```suggestion
The Background Synchronization API provides a way to defer tasks to be run in a {{domxref('Service Worker API','service worker')}} until the user has a stable network connection.
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+## Web Background Synchronization Concepts and Usage</h2>
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:21]:
```suggestion
## Background Synchronization Concepts and Usage
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+For example, an email client application could let its users compose and send messages at any time, even when the device has no network connection. The application frontend just registers a sync request and the service worker gets alerted when the network is present again and handles the sync.
+
+The {{domxref('SyncManager')}} interface is available through {{domxref('ServiceWorkerRegistration.sync')}}. A unique tag identifier is set to 'name' the sync event, which can then be listened for within the {{domxref('ServiceWorker')}} script. Once the event is received you can then run any functionality available, such as sending requests to the server.
+
+As this API relies on service workers, functionality provided by this API is only available in a secure context.
+
+> **Note:** At the time of writing, the Web Background Synchronization API is only available through an installed [Progressive Web App](/en-US/docs/Web/Progressive_web_apps)
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:31]:
```suggestion
> **Note:** At the time of writing, the Background Synchronization API is only available through an installed [Progressive Web App](/en-US/docs/Web/Progressive_web_apps)
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+## Web Background Synchronization Interfaces
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:33]:
```suggestion
## Background Synchronization Interfaces
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+- {{domxref('SyncManager')}}
+  - : Registers tasks to be run in a service worker at a later time with network connectivity. These tasks are referred to as background sync requests.
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:36]:
```suggestion
  - : Registers tasks to be run in a service worker at a later time with network connectivity. These tasks are referred to as _background sync requests_.
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  - : Represents a synchronization event, sent to the {{domxref('ServiceWorkerGlobalScope', 'global scope')}} of a {{domxref('ServiceWorker')}}. It provides a way to run tasks in the service worker with network connectivity.
+
+## Service Worker Additions
+
+The following additions to the {{domxref('Service Worker API')}} are specified in the Background Sync specification to provide an entry point for using Background Sync.
+
+- {{domxref("ServiceWorkerRegistration.sync")}} {{readonlyinline}}
+  - : Returns a reference to the {{domxref("SyncManager")}} interface for registering tasks to run at specific intervals.
+- {{domxref("ServiceWorkerGlobalScope.onsync")}}
+  - : An event handler fired whenever a {{Event("sync")}} event occurs. This happens at timed intervals, that are specified when registering a {{domxref('SyncManager')}}.
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:47]:
```suggestion
  - : An event handler fired whenever a {{Event("sync")}} event occurs. This happens at timed intervals that are specified when registering a {{domxref('SyncManager')}}.
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    await registration.sync.register('sync-messages');
+  } catch {
+    console.log('Background Sync could not be registered!');
+  }
+}
+```
+
+### Verifying a Background Sync by Tag
+
+This code checks to see if a Background Sync task with a given tag is registered.
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:70]:
```suggestion
This code checks to see if a background-sync task with a given tag is registered.
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+```js
+navigator.serviceWorker.ready.then(registration => {
+  registration.sync.getTags().then(tags => {
+    if (tags.includes('sync-messages'))
+      console.log('Messages sync already requested');
+  });
+});
+```
+
+### Listening for a Background Sync within a Service Worker
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:81]:
```suggestion
### Listening for a background sync within a Service Worker
```

File: files/en-us/web/api/web_background_synchronization_api/index.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+The following example shows how to respond to a background sync event in the service worker.
```
Comment (sideshowbarker) [inline files/en-us/web/api/web_background_synchronization_api/index.md:83]:
```suggestion
The following example shows how to respond to a background-sync event in the service worker.
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]
1. I recognize that the spec title is “Web Background Synchronization”, but for consistency with how we handle other cases, we should drop the “Web” part in MDN — including in the `files/en-us/web/api/web_background_synchronization_api` directory name (so, would need to do a `git mv` on the branch.

2. The `files/en-us/web/api/serviceworkerglobalscope/index.html` changes are fine but in a couple of hours, that `index.html` isn’t going to exist in the repo any longer, so those changes will need to be replicated in the `index.md` source (so this PR branch will need to be rebased after the big markdown-conversion merge happens today

File: files/en-us/web/api/web_background_synchronization_api/index.md (hunk 8 above)
Comment (captainbrosset) [reply files/en-us/web/api/web_background_synchronization_api/index.md]: This was actually the incorrect definition. I ended up changing the whole line.

File: files/en-us/web/api/web_background_synchronization_api/index.md (hunk 9 above)
Comment (captainbrosset) [reply files/en-us/web/api/web_background_synchronization_api/index.md]: I didn't make these changes, and kept `background sync` (with no dash). English isn't my first language, but it felt odd to me linking those 2 words like this. Let me know if this is needed.

**Review by sideshowbarker — APPROVED** [review]

**Comment by captainbrosset** [comment]
Woohoo, first PR merged! Thank you for the really quick review and useful info on the chat. A very smooth experience!

**Comment by sideshowbarker** [comment]
Cheers 🎉 Thanks and congrats on getting this landed — it’s a great addition

**Pull request merged**